server/sonar-server/src/main/java/org/sonar/server/settings/ws/ValuesAction.java

```diff
@@ -0,0 +1,213 @@
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.apache.commons.lang.ObjectUtils;
+import org.sonar.api.config.PropertyDefinition;
+import org.sonar.api.config.PropertyDefinitions;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.component.ComponentDto;
+import org.sonar.db.property.PropertyDto;
+import org.sonarqube.ws.Settings;
+import org.sonarqube.ws.Settings.ValuesWsResponse;
+
+import static org.elasticsearch.common.Strings.isNullOrEmpty;
+import static org.sonar.api.PropertyType.PROPERTY_SET;
+import static org.sonar.server.settings.ws.SettingsWsComponentParameters.addComponentParameters;
+import static org.sonar.server.ws.WsUtils.writeProtobuf;
+import static org.sonarqube.ws.client.setting.SettingsWsParameters.ACTION_VALUES;
+import static org.sonarqube.ws.client.setting.SettingsWsParameters.PARAM_COMPONENT_ID;
+import static org.sonarqube.ws.client.setting.SettingsWsParameters.PARAM_COMPONENT_KEY;
+import static org.sonarqube.ws.client.setting.SettingsWsParameters.PARAM_KEYS;
+
+public class ValuesAction implements SettingsWsAction {
+
+  private static final Splitter DOT_SPLITTER = Splitter.on(".").omitEmptyStrings();
+  private static final Splitter COMMA_SPLITTER = Splitter.on(",");
+
+  private final DbClient dbClient;
+  private final SettingsWsComponentParameters settingsWsComponentParameters;
+  private final PropertyDefinitions propertyDefinitions;
+  private final SettingsLoader settingsLoader;
+
+  public ValuesAction(DbClient dbClient, SettingsWsComponentParameters settingsWsComponentParameters, PropertyDefinitions propertyDefinitions, SettingsLoader settingsLoader) {
+    this.dbClient = dbClient;
+    this.settingsWsComponentParameters = settingsWsComponentParameters;
+    this.propertyDefinitions = propertyDefinitions;
+    this.settingsLoader = settingsLoader;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context.createAction(ACTION_VALUES)
+      .setDescription("List settings values.<br>" +
+        "If no value have been set for a setting, then the default value is returned.<br>" +
+        "Either '%s' or '%s' can be provided, not both.<br> " +
+        "Requires one of the following permissions: " +
+        "<ul>" +
+        "<li>'Administer System'</li>" +
+        "<li>'Administer' rights on the specified component</li>" +
+        "</ul>", PARAM_COMPONENT_ID, PARAM_COMPONENT_KEY)
+      .setResponseExample(getClass().getResource("values-example.json"))
+      .setSince("6.1")
+      .setHandler(this);
+    addComponentParameters(action);
+    action.createParam(PARAM_KEYS)
```

Comment:
we want to handle multiple value parameter now. It makes sense to use the singular for multi value parameter. Then either you use `key=firstKey&key=secondKey` or you use `key=firstKey,secondKey` and we privilege the first option from now on.